vsphere/virtual_machine_config_structure.go

```diff
@@ -0,0 +1,664 @@
+// reboot_required to true.
+func getWithRestart(d *schema.ResourceData, key string) interface{} {
+	if d.HasChange(key) {
+		log.Printf("[DEBUG] %s: Resource argument %q requires a VM restart", resourceVSphereVirtualMachineIDString(d), key)
+		d.Set("reboot_required", true)
+	}
+	return d.Get(key)
+}
+
+// getBoolWithRestart fetches a *bool for the resource data item specified at
+// key. If the value has changed, a reboot is flagged in the virtual machine by
+// setting reboot_required to true.
+//
+// This function always returns at least false, even if a value is unspecified.
+func getBoolWithRestart(d *schema.ResourceData, key string) *bool {
+	if d.HasChange(key) {
+		d.Set("reboot_required", true)
+	}
+	return structure.GetBool(d, key)
+}
+
+// schemaVirtualMachineConfigSpec returns schema items for resources that
+// need to work with a VirtualMachineConfigSpec.
+func schemaVirtualMachineConfigSpec() map[string]*schema.Schema {
+	s := map[string]*schema.Schema{
+		// VirtualMachineBootOptions
+		"boot_delay": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "The number of milliseconds to wait before starting the boot sequence.",
+		},
+		"efi_secure_boot_enabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "When the boot type set in firmware is efi, this enables EFI secure boot.",
+		},
+		"boot_retry_delay": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Default:     10000,
+			Description: "The number of milliseconds to wait before retrying the boot sequence. This only valid if boot_retry_enabled is true.",
+		},
+		"boot_retry_enabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "If set to true, a virtual machine that fails to boot will try again after the delay defined in boot_retry_delay.",
+		},
+
+		// VirtualMachineFlagInfo
+		"enable_disk_uuid": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Expose the UUIDs of attached virtual disks to the virtual machine, allowing access to them in the guest.",
+		},
+		"hv_mode": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Default:      string(types.VirtualMachineFlagInfoVirtualExecUsageHvAuto),
+			Description:  "The (non-nested) hardware virtualization setting for this virtual machine. Can be one of hvAuto, hvOn, or hvOff.",
+			ValidateFunc: validation.StringInSlice(virtualMachineVirtualExecUsageAllowedValues, false),
+		},
+		"ept_rvi_mode": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Default:      string(types.VirtualMachineFlagInfoVirtualMmuUsageAutomatic),
+			Description:  "The EPT/RVI (hardware memory virtualization) setting for this virtual machine. Can be one of automatic, on, or off.",
+			ValidateFunc: validation.StringInSlice(virtualMachineVirtualMmuUsageAllowedValues, false),
+		},
+		"enable_logging": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
```

Comment:
without knowing exactly what's being logged - I'd question this being defaulted to `true` (but you know this better than me 😄)

Reply:
Yeah, on second thought I'm switching this back to `false`. I don't think anything sensitive is logged here, but better safe than sorry.